File: frontend/e-commerce-frontend/src/app/shared/models/customer-profile.ts
export enum MembershipType {
  STANDARD = 'STANDARD',
  PREMIUM = 'PREMIUM',
  VIP = 'VIP'
}

export interface DtoCustomerProfile {
  id?: number;
  userId?: number;
  age?: number;
  city?: string;
  state?: string;
  country?: string;
  membershipType?: MembershipType;
}

export interface DtoCustomerProfileRequest {
  age?: number;
  city?: string;
  state?: string;
  country?: string;
}

File: backend/ecommerce-backend/src/main/java/com/furkan/entities/CustomerProfile.java
package com.furkan.entities;

import com.furkan.enums.MembershipType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "customer_profiles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class CustomerProfile extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private int age;

    @Column(length = 100)
    private String city;

    @Column(length = 100)
    private String state;

    @Column(length = 50)
    private String country;

    @Column(length = 50)
    private MembershipType membershipType = MembershipType.STANDARD; // Örn: STANDARD, PREMIUM, VIP

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", referencedColumnName = "id", nullable = false)
    private User user;
}
